### PR TITLE
[FRON-1536]: Refactored the code. Added a logic to reverse the cancel…

### PR DESCRIPTION
**Pull request by aashishgurung**
#### 1. Objective

Update the order status when charge status is updated manually by Operations Team.

Jira Ticket: [#1536](https://opn-ooo.atlassian.net/browse/SHOPIFYAPP-247)

#### 2. Description of change

Added a logic to reverse the cancelled order and set the status as processing when the order status is cancelled and the charge is successful. This will be triggered via the webhook. Refactored the `Complete.php` file.

#### 3. Quality assurance

- Checkout with any item
- On reaching the authorize page,
   - First, test by authorizing the payment
   - Second, test by not authorizing the payment. Simply close the page once you reach there.
- On Magento admin, find the order and cancel it manually.
   - You can cancel the order from Omise admin dashboard by marking the charge as failed.
- On Omise admin dashboard, find the charge and mark it as a success.
- On Magento admin, check the order you just cancelled. It status should be set to processing and appropriate comment should be there in the comment history.

**🔧 Environments:**

- Platform version: Magento 2.4.4
- Omise plugin version: Omise-Magento 2.26.0
- PHP version: 7.4.28

### CODE REVIEW

File: Plugin/ConfigSectionPaymentPlugin.php [hunk 1]
```diff
@@ -69,8 +69,12 @@ private function getKeys($configData)
     {
         $configFields = $configData['groups']['omise']['fields'];
 
-        $publicKeyIndex = ($configFields['sandbox_status']) ? 'test_public_key' : 'live_public_key';
-        $secretKeyIndex = ($configFields['sandbox_status']) ? 'test_secret_key' : 'live_secret_key';
+        $standboxStatus = array_key_exists('value', $configFields['sandbox_status'])
```
Comment (hpariwat) [inline Plugin/ConfigSectionPaymentPlugin.php:72]:
```suggestion
        $sandboxStatus = array_key_exists('value', $configFields['sandbox_status'])
```
I guess you meant `$sandboxStatus`.

Comment (aashishgurung) [reply Plugin/ConfigSectionPaymentPlugin.php:72]: @hpariwat Yes, thanks.

**Review by kan98 — APPROVED** [review]

File: Plugin/ConfigSectionPaymentPlugin.php [hunk 2]
```diff
@@ -69,8 +69,12 @@ private function getKeys($configData)
     {
         $configFields = $configData['groups']['omise']['fields'];
 
-        $publicKeyIndex = ($configFields['sandbox_status']) ? 'test_public_key' : 'live_public_key';
-        $secretKeyIndex = ($configFields['sandbox_status']) ? 'test_secret_key' : 'live_secret_key';
+        $sandboxStatus = array_key_exists('value', $configFields['sandbox_status'])
```
Comment (new4762) [inline Plugin/ConfigSectionPaymentPlugin.php:72]: ah i see you will update it but 
i think for got about this comment 
```php
 // if sandbox status is updated the updated value will be under 'value' key else it won't have the value key
 ```

**Review by new4762 — APPROVED** [review]

File: Plugin/ConfigSectionPaymentPlugin.php (hunk 2 above)
Comment (aashishgurung) [reply Plugin/ConfigSectionPaymentPlugin.php:72]: @new4762 Thanks.

**Review by hpariwat — APPROVED** [review]

**Review by tookaa2 — APPROVED** [review]

**Pull request merged**